akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -820,6 +820,12 @@ private[akka] class ActorSystemImpl(
 
   private val _dynamicAccess: DynamicAccess = createDynamicAccess()
 
+  /**
+   * Optimistic optimization: Typed actors will _all_ pick up the system adapter, going through
+   * extension infra only once per thread
+   */
```

Comment:
Make the comment more clear about what type the `AnyRef` has. Also, I find the "once per thread" misleading, reading it like a thread local.